Problem Running Property Interrogator Tool

While trying to run the Property Interrogator Tool for the RO model in Proteuslib, I found that I could no longer use the tool successfully. 

Here is an example of the implementation:
```
from pyomo.environ import ConcreteModel, units
from idaes.core import FlowsheetBlock, Solute, Solvent, LiquidPhase, Phase
from proteuslib.unit_models.reverse_osmosis_0D import ReverseOsmosis0D
from idaes.generic_models.properties.interrogator import PropertyInterrogatorBlock

m = ConcreteModel()
m.fs = FlowsheetBlock(default={"dynamic": False, "time_units": units.s})
m.fs.properties = PropertyInterrogatorBlock(default={
        "phase_list": {"Liq": LiquidPhase},
        "component_list": {"H2O": Solvent,
                           "NaCl": Solute}})
m.fs.unit = ReverseOsmosis0D(default={
    "property_package": m.fs.properties})
```
Here is the error that I get :
```
Traceback (most recent call last):
  File "C:/Users/adama/my-idaes-dev/idaes-pse/SANDBOX/SANDBOX/sim_prop_interrogator.py", line 27, in <module>
    m.fs.properties = PropertyInterrogatorBlock(default={
  File "C:\Users\adama\Anaconda3\envs\my-idaes-dev\lib\site-packages\pyomo\core\base\block.py", line 542, in __setattr__
    self.add_component(name, val)
  File "C:\Users\adama\Anaconda3\envs\my-idaes-dev\lib\site-packages\pyomo\core\base\block.py", line 1087, in add_component
    val.construct(data)
  File "C:\Users\adama\Anaconda3\envs\my-idaes-dev\lib\site-packages\pyomo\core\base\block.py", line 1984, in construct
2021-07-19 13:31:40 [ERROR] idaes.core.process_block: Failure in build: fs.properties
Traceback (most recent call last):
  File "C:\Users\adama\Anaconda3\envs\my-idaes-dev\lib\site-packages\idaes\core\process_block.py", line 37, in _rule_default
    b.build()
  File "C:\Users\adama\Anaconda3\envs\my-idaes-dev\lib\site-packages\idaes\generic_models\properties\interrogator\properties_interrogator.py", line 82, in build
    raise Exception()
Exception
ERROR: Constructing component 'fs.properties' from data=None failed:
    Exception:
    self._getitem_when_not_present(_idx)
  File "C:\Users\adama\Anaconda3\envs\my-idaes-dev\lib\site-packages\pyomo\core\base\block.py", line 1909, in _getitem_when_not_present
    obj = self._rule(_block, idx)
  File "C:\Users\adama\Anaconda3\envs\my-idaes-dev\lib\site-packages\pyomo\core\base\util.py", line 313, in __call__
    return self._fcn(parent, idx)
  File "C:\Users\adama\Anaconda3\envs\my-idaes-dev\lib\site-packages\idaes\core\process_block.py", line 37, in _rule_default
    b.build()
  File "C:\Users\adama\Anaconda3\envs\my-idaes-dev\lib\site-packages\idaes\generic_models\properties\interrogator\properties_interrogator.py", line 82, in build
    raise Exception()
Exception
```
I think the problem has something to do with my environment, but the `Exception` may need to be given a type and relevant error message.

--- a/idaes/generic_models/properties/interrogator/properties_interrogator.py
+++ b/idaes/generic_models/properties/interrogator/properties_interrogator.py
@@ -31,9 +31,9 @@ from idaes.core import (declare_process_block_class,
                         UnitModelBlockData,
                         Phase,
                         LiquidPhase,
-                        SolidPhase,
                         VaporPhase,
                         Component)
+from idaes.core.util.exceptions import ConfigurationError
 import idaes.logger as idaeslog
 
 # Some more information about this module
@@ -78,9 +78,10 @@ class PropertyInterrogatorData(PhysicalParameterBlock):
             for p, t in self.config.phase_list.items():
                 if t is None:
                     t = Phase
-                elif not issubclass(
-                        t, Phase):
-                    raise Exception()
+                elif not issubclass(t, Phase):
+                    raise ConfigurationError(
+                        f"{self.name} invalid phase type {t} (for phase {p})."
+                        f" Type must be a subclass of Phase.")
                 self.add_component(p, t())
 
         # Component objects
@@ -92,7 +93,10 @@ class PropertyInterrogatorData(PhysicalParameterBlock):
                 if t is None:
                     t = Component
                 elif not issubclass(t, Component):
-                    raise Exception()
+                    raise ConfigurationError(
+                        f"{self.name} invalid component type {t} (for "
+                        f"component {j}). Type must be a subclass of "
+                        f"Component.")
                 self.add_component(j, t())
 
         # Set up dict to record property calls

--- a/idaes/generic_models/properties/interrogator/properties_interrogator.py
+++ b/idaes/generic_models/properties/interrogator/properties_interrogator.py
@@ -15,6 +15,7 @@ Tool to interrogate IDAES flowsheets and list the physical properties
 required to simulate it.
 """
 import sys
+from inspect import isclass
 
 # Import Pyomo libraries
 from pyomo.environ import Set, Var, units as pyunits
@@ -78,7 +79,7 @@ class PropertyInterrogatorData(PhysicalParameterBlock):
             for p, t in self.config.phase_list.items():
                 if t is None:
                     t = Phase
-                elif not issubclass(t, Phase):
+                elif not isclass(t) or not issubclass(t, Phase):
                     raise ConfigurationError(
                         f"{self.name} invalid phase type {t} (for phase {p})."
                         f" Type must be a subclass of Phase.")
@@ -92,7 +93,7 @@ class PropertyInterrogatorData(PhysicalParameterBlock):
             for j, t in self.config.component_list.items():
                 if t is None:
                     t = Component
-                elif not issubclass(t, Component):
+                elif not isclass(t) or not issubclass(t, Component):
                     raise ConfigurationError(
                         f"{self.name} invalid component type {t} (for "
                         f"component {j}). Type must be a subclass of "

--- a/idaes/generic_models/properties/interrogator/tests/test_properties_interrogator.py
+++ b/idaes/generic_models/properties/interrogator/tests/test_properties_interrogator.py
@@ -30,6 +30,7 @@ from idaes.generic_models.unit_models.separator import \
     Separator, SplittingType
 from idaes.generic_models.properties.interrogator import \
     PropertyInterrogatorBlock
+from idaes.core.util.exceptions import ConfigurationError
 
 
 @pytest.mark.unit
@@ -518,3 +519,28 @@ def test_interrogator_state_block_methods_custom_phase_comps():
             "material density terms": ["fs.props"],
             "enthalpy flow terms": ["fs.props"],
             "energy density terms": ["fs.props"]}
+
+
+@pytest.mark.unit
+def test_interrogator_parameter_block_custom_phase_error():
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(default={"dynamic": False})
+
+    with pytest.raises(ConfigurationError,
+                       match="fs.params invalid phase type foo \(for phase "
+                       "P1\). Type must be a subclass of Phase."):
+        m.fs.params = PropertyInterrogatorBlock(default={
+            "phase_list": {"P1": "foo", "P2": None}})
+
+
+@pytest.mark.unit
+def test_interrogator_parameter_block_custom_comp_error():
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(default={"dynamic": False})
+
+    with pytest.raises(ConfigurationError,
+                       match="fs.params invalid component type foo \(for "
+                       "component c1\). Type must be a subclass of "
+                       "Component."):
+        m.fs.params = PropertyInterrogatorBlock(default={
+            "component_list": {"c1": "foo", "c2": None}})

